src/hotspot/share/asm/assembler.hpp

```diff
@@ -290,6 +290,18 @@ class AbstractAssembler : public ResourceObj  {
   };
 #endif
 
+  // sign-extended tolerant cast needed by callers of emit_int8 and emit_int16
+  template <typename T>
+  constexpr T narrow_cast(int x) const {
+   if (x < 0) {
+    assert(x > -(std::numeric_limits<T>::max() - 1), "too negative");
+    return static_cast<T>(x);  // cut off sign bits
+   } else {
+     return checked_cast<T>(x);
+   }
+  }
+
+
  public:
```

Comment:
I don't think we need this.  See below.

Reply:
Nevermind, I tried my alternative idea below and it didn't work.  For these particular cases where we only care about going to uint8_t, we could check is8bit().  Another trick I've seen is checking if (val >> width) is 0 or -1.

Reply:
There is an is8bit() test that precedes these casts.